src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -387,20 +389,36 @@ public UUID getFirstSourceDatasetIdFromSnapshotId(
   }
 
   public SnapshotPreviewModel retrievePreview(
-      UUID snapshotId, String tableName, int limit, int offset) {
+      UUID snapshotId,
+      String tableName,
+      int limit,
+      int offset,
+      String sort,
+      SqlSortDirection direction) {
     Snapshot snapshot = retrieve(snapshotId);
 
-    snapshot.getTables().stream()
-        .filter(t -> t.getName().equals(tableName))
-        .findFirst()
-        .orElseThrow(
-            () ->
-                new SnapshotPreviewException(
-                    "No snapshot table exists with the name: " + tableName));
+    SnapshotTable table =
+        snapshot
+            .getTableByName(tableName)
+            .orElseThrow(
+                () ->
+                    new SnapshotPreviewException(
+                        "No snapshot table exists with the name: " + tableName));
+
+    SqlSortDirection sortDirection = Optional.ofNullable(direction).orElse(SqlSortDirection.ASC);
```

Comment:
No need for `Optional` here
```suggestion
    SqlSortDirection sortDirection = Objects.requireNonNullElse(direction, SqlSortDirection.ASC);
```